traversal/procedure/GraphProcedure.java

```diff
@@ -183,10 +185,29 @@ public Builder() {
         }
 
         public GraphProcedure build() {
+            linearise();
```

Comment:
note that we actually have a procedure permutation generator, that generates all possible permutations of a query procedure using the `Builder`. However, now that we have this, that testing framework won't be as effective...
I recommend making two APIs: one called `lineariseAndBuild` and `build`. The `linearseAndBuild` should be called from the engine, and the `build` can be used from the test.

Reply:
let's do this whole operation in the `GraphPlanner` actually!